quodlibet/quodlibet/formats/_audio.py

```diff
@@ -536,24 +538,104 @@ def _role_call(self, role_tag, sub_keys=None):
+        lyric_filenames = \
+            config.getstringlist("memory", "lyric_filenames", [])
+        # ensure some default pathfile names
+        lyric_filenames.append(
+            sanitise(os.sep, [(self.comma("lyricist") or
+                              self.comma("artist")),
+                              self.comma("title")]) + u'.lyric')
+        lyric_filenames.append(
+            sanitise(' - ', [(self.comma("lyricist") or
+                             self.comma("artist")),
+                             self.comma("title")]) + u'.lyric')
+
+        # generate all potential paths (unresolved/unexpanded)
+        pathfiles = OrderedDict()
+        for p in lyric_paths:
+            for f in lyric_filenames:
+                pathfile = ""
+                if os.name == "nt":
+                    # this was added at a later point. only use
+                    # escape_filename here to keep the linux case
+                    # the same as before
+                    pathfile = \
+                        os.path.join(p, os.path.dirname(f),
+                                     escape_filename(os.path.basename(f)))
+                else:
+                    pathfile = \
+                        os.path.join(p, os.path.dirname(f),
+                                    fsnative(os.path.basename(f)))
+                if not pathfile in pathfiles:
+                    pathfiles[pathfile] = pathfile
+
+        #print_d("searching for lyrics in:\n%s" % '\n'.join(pathfiles.keys()))
+
+        # expand each raw pathfile in turn and test for existence
+        match_ = ""
+        pathfiles_expanded = OrderedDict()
+        rx_params = re.compile('[^\\\]<[^' + re.escape(os.sep) + ']*[^\\\]>')
+        for pathfile in pathfiles.keys():
+            path = expanduser(pathfile)
+            if rx_params.search(path):
+                path = ArbitraryExtensionFileFromPattern(pathfile).format(self)
+            pathfiles_expanded[path] = path  # resolved as late as possible
+            if os.path.exists(path):
+                match_ = path
+                break
+
+        if not match_:
+            # search even harder!
+            lyric_extensions = ['lyric', 'lyrics', '', 'txt']
+            #print_d("extending search to extensions: %s" % lyric_extensions)
+
+            def generate_mod_ext_paths(pathfile):
+                # separate pathfile's extension (if any)
+                ext = os.path.splitext(pathfile)[1][1:]
+                path = pathfile[:-1 * len(ext)].strip('.') if ext else pathfile
+                # skip the proposed lyric extension if it is the same as
+                # the original for a given search pathfile stub - it has
+                # already been tested without success!
+                extra_extensions = [x for x in lyric_extensions if x != ext]
+
+                # join valid new extensions to pathfile stub and return
+                return list(map(lambda ext: '.'.join([path, ext])
```

Comment:
Also comprehension would read nicely here: `['.'.join([path, ext] if ext else path for ext in extra_extensions]`, or something.

Reply:
done